home/tests/test_serializers.py

```diff
@@ -0,0 +1,388 @@
+    format_whatsapp_message,
+    has_next_message,
+    has_previous_message,
+)
+
+from .page_builder import (
+    MBlk,
+    MBody,
+    PageBuilder,
+    SBlk,
+    SBody,
+    UBlk,
+    UBody,
+    VarMsg,
+    VBlk,
+    VBody,
+    WABlk,
+    WABody,
+)
+
+
+@pytest.mark.django_db
+class TestHasNextMessage:
+    def create_content_page(
+        self,
+        parent: Union[ContentPage, None] = None,
+        title: str = "default page",
+        tags: Union[List[str], None] = None,
+        wa_body_count: int = 1,
+        publish: bool = True,
+    ) -> ContentPage:
+        """
+        Helper function to create pages needed for each test.
+
+        Parameters
+        ----------
+        parent : ContentPage
+            The ContentPage that will be used as the parent of the content page.
+
+            If this is not provided, a ContentPageIndex object is created as a child of
+            the default home page and that is used as the parent.
+        title : str
+            Title of the content page.
+        tags : [str]
+            List of tags on the content page.
+        wa_body_count : int
+            How many WhatsApp message bodies to create on the content page.
+        publish: bool
+            Should the content page be published or not.
+        """
+        if not parent:
+            home_page = HomePage.objects.first()
+            main_menu = PageBuilder.build_cpi(home_page, "main-menu", "Main Menu")
+            parent = main_menu
+
+        bodies = [
+            MBody(title, [MBlk("*Default Messenger Content* 🏥")]),
+            SBody(title, [SBlk("*Default SMS Content* ")]),
+            UBody(title, [UBlk("*Default USSD Content* ")]),
+            VBody(title, [VBlk("*Default Viber Content* ")]),
+        ]
+
+        for i in range(wa_body_count):
+            wa_body = f"*Default WhatsApp Content {i+1}* 🏥"
+            bodies.append(WABody(title, [WABlk(wa_body)]))
+
+        content_page = PageBuilder.build_cp(
+            parent=parent,
+            slug=title.replace(" ", "-"),
+            title=title,
+            bodies=bodies,
+            tags=tags or [],
+            quick_replies=[],
+            triggers=[],
+            publish=publish,
+        )
+        return content_page
+
+    # Test has_next_message
+    def test_has_next_message_whatsapp(self) -> None:
+        """Checks if the next message matches up.
+        Index of first message is 0, next message is then message 2 with index 1"""
+        page = self.create_content_page(wa_body_count=2)
+        assert page.whatsapp_body
+        assert has_next_message(0, page, "whatsapp") == 2
+
+    def test_has_next_message_viber_no_next_message(self) -> None:
+        """On a page with 2 viber messages, the next message is None"""
+        page = self.create_content_page()
+        assert page.viber_body
+        assert has_next_message(1, page, "viber") is None
+
+    def test_has_next_message_invalid_platform(self) -> None:
```

Comment:
We don't need `has_next_message` in the test name, because that's already in the class name. Perhaps `test_no_next_message_for_invalid_platform` would be a better name here?

(Similarly for various other test names.)